packages/next/build/index.ts

```diff
@@ -511,27 +511,27 @@ export default async function build(
     const webpackBuildStart = process.hrtime()
 
     let result: CompilerResult = { warnings: [], errors: [] }
-    // TODO: why do we need this?? https://github.com/vercel/next.js/issues/8253
-    if (isLikeServerless) {
-      const clientResult = await runCompiler(clientConfig)
-      // Fail build if clientResult contains errors
-      if (clientResult.errors.length > 0) {
-        result = {
-          warnings: [...clientResult.warnings],
-          errors: [...clientResult.errors],
-        }
-      } else {
-        const serverResult = await runCompiler(configs[1])
-        result = {
-          warnings: [...clientResult.warnings, ...serverResult.warnings],
-          errors: [...clientResult.errors, ...serverResult.errors],
+    // we run client and server compilation separately and for
```

Comment:
Do we want to default to this behavior, or only to surface an option? This could slow down builds that aren't facing memory constraints.

Reply:
Updated to maintain current behavior as discussed